tests/utils.py

```diff
@@ -203,13 +204,15 @@ def upload_file(file_path):
     return s3.http_url(basename)
 
 
-def submit_job(app_url, app_args, app_name="/spark", args=[]):
+def submit_job(app_url, app_args, app_name="/spark", args=[], args_str=None):
     if is_strict():
```

Comment:
The handling of the arguments is getting pretty complex now with the addition of the extra keyword parameters (see my comment above as to why this may not be required). We could also restructure this function as:
```python
def submit_job(app_url, app_args, app_name="/spark", args=[]):

    submit_args = []
    if is_strict():
        submit_args += ["--conf", 'spark.mesos.driverEnv.MESOS_MODULES=file:///opt/mesosphere/etc/mesos-scheduler-modules/dcos_authenticatee_module.json']
        submit_args += ["--conf", 'spark.mesos.driverEnv.MESOS_AUTHENTICATEE=com_mesosphere_dcos_ClassicRPCAuthenticatee']
        submit_args += ["--conf", 'spark.mesos.principal=service-acct']

    submit_args += args
    submit_args += ["--conf", "spark.driver.memory=2g", app_url]

    if isinstance(app_args, str):
        # Assuming app_args is a string
        submit_args += [app_args, ]
    else:
        # Assuming that app_args is a list
        submit_args += app_args

    # We could also change the following to `s for s in submit_args if s is not None`
    submit_args_string = " ".join(s for s in submit_args)

    cmd = ["dcos", "spark", "--name={}".format(app_name), "run", "--verbose", ]
    cmd += ['--submit-args="{}"'.format(submit_args_string), ]

    LOGGER.info("Running %s", cmd)
    # Note that `subprocess` allows command to be specified as a list as well as a string.
    stdout = subprocess.check_output(cmd, shell=True).decode('utf-8')

    LOGGER.info("stdout: %s", stdout)

    regex = r"Submission id: (\S+)"
    match = re.search(regex, stdout)
    return match.group(1)
```
Where the handling of each of the different argument types: strict, default, and user-specified; are now more clearly distinguishable.

Reply:
Instead of doing this, I'll omit the extra parameter.